src/xdp/extensionset.c

```diff
@@ -94,6 +95,8 @@ XdpExtensionSetAssignLayout(
     UINT8 MaxAlignment = BaseAlignment;
     UINT8 CurrentAlignment;
 
+    FRE_ASSERT(!ExtensionSet->LayoutAssigned);
```

Comment:
Why did you choose a free assert instead of a debug?

Reply:
This module includes enforcement of the native XDP driver API, so these FRE_ASSERTs are designed to detect driver misbehavior on any system, not just with CHK builds. It's not a performance-sensitive path.

Reply:
If you had support for driver verifier, would you enable these only if driver verifier was enabled? QUIC has 4 types of assert validation:

- Debug
- Telemetry
- Free
- Verifier

We haven't made a huge distinction, but the verifier ones are nice if the assert is about external conformance.

Reply:
Agreed, we would ideally have a separate verifier mode. Right now we just have debug/release builds, plus a knob to make the debug build also inject random failures.